onnxruntime/contrib_ops/cpu/bert/attention_cpu_base.h

```diff
@@ -102,10 +103,17 @@ class AttentionCPUBase : public AttentionBase {
     }
 
     // Compute the attention score.
-    size_t bytes = SafeInt<size_t>(batch_size) * num_heads_ * sequence_length * total_sequence_length * sizeof(T);
-    auto attention_probs = allocator->Alloc(bytes);
+    void* attention_probs = nullptr;
+    T* attn_probs_data = nullptr;
+    if (attn_probs == nullptr) {
+      size_t bytes = SafeInt<size_t>(batch_size) * num_heads_ * sequence_length * total_sequence_length * sizeof(T);
+      attention_probs = allocator->Alloc(bytes);
```

Comment:
There is no need to allocate extra space if we do not output it. The original implementation has handled the temp space allocation.